cdm/core/src/test/java/ucar/nc2/util/cache/TestNetcdfFileCache.java

```diff
@@ -148,6 +159,36 @@ public void testNetcdfFileCache() throws IOException {
     }
 
     cache.clearCache(true);
+
+    // Verify that cache hits actually happen.
+    // Load up the files and then close them (thus, unlocking them).
+    // Then load them up again.
+    // Since files are now unlocked, loading stuff up should result in hits.
+    loadFilesIntoCache(new File(TestDir.cdmLocalTestDataDir), cache);
+    assertThat(cache.hits.get()).isEqualTo(0);
+    assertThat(cache.miss.get()).isEqualTo(6 * saveCount);
+    map = cache.getCache();
+    // close all
+    files = new ArrayList<>();
+    for (Object key : map.keySet()) {
+      FileCache.CacheElement elem = map.get(key);
+      assertThat(elem.list.size()).isEqualTo(1);
+      for (FileCache.CacheElement.CacheFile file : elem.list) {
+        // Need to do it this way instead of directly closing them
+        // because closing the files changes the iterations
+        files.add(file.ncfile);
+      }
+    }
+    for (FileCacheable ncfile : files) {
+      ncfile.close();
+    }
+
+    loadFilesIntoCache(new File(TestDir.cdmLocalTestDataDir), cache);
+    cache.showCache(new Formatter(System.out));
+    assertThat(cache.hits.get()).isEqualTo(saveCount);
+    assertThat(cache.miss.get()).isEqualTo(6 * saveCount);
```

Comment:
Alright, so what I think is happening here is that there is no guarantee at this point that all of the files are getting marked as unlocked before we try to reload the files into the cache, so then the hit count isn't what I'd expect. 

Any suggestions like a possible cache method call or using a short timer delay?